cmake/modules/OmrDetectSystemInformation.cmake

```diff
@@ -193,6 +193,17 @@ macro(omr_detect_system_information)
 			if("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC" OR "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
 				# clang on Windows mimics MSVC
 				set(_OMR_TOOLCONFIG "msvc")
+			elseif(CMAKE_C_COMPILER_ID MATCHES "^Clang$")
+				# OpenXL17 uses CMAKE_C_COMPILER_ID "Clang" for AIX
+				set(_OMR_TOOLCONFIG "gnu")
+				set(CMAKE_C_COMPILER_IS_OPENXL TRUE CACHE BOOL "OpenXL is the C compiler")
+				if(CMAKE_C_COMPILER_IS_OPENXL)
+					set(OMR_ENV_OPENXL 1)
+					set(ENV{OMR_ENV_OPENXL} ${OMR_ENV_OPENXL})
+				else()
+					set(OMR_ENV_OPENXL 0)
+					set(ENV{OMR_ENV_OPENXL} ${OMR_ENV_OPENXL})
+				endif()
```

Comment:
Why would we want this? Won't `CMAKE_C_COMPILER_IS_OPENXL` be sufficient?

Reply:
That environment variable is for Makefiles to check if the compiler is openxl. I used that env variable instead of adding openxl in defaults.yml

Reply:
Which makefiles? With cmake, we don't write makefiles (cmake might even use another generator, e.g. `ninja`).
I don't see any references to `OMR_ENV_OPENXL`, neither in cmakefiles nor the environment; perhaps it would help if you provided a sample of how you imagine they would be used.

Reply:
Yes. I used this CMAKE variable to let the makefiles know that the compiler is openxl. The other way of doing it may be by setting the compiler in defaults.yml

Reply:
My point is that when using `cmake`, files like `openj9/runtime/gc_glue_java/configure_includes/configure_aix_ppc.mk` are not used, so the reference there (in https://github.com/eclipse-openj9/openj9/pull/20045) is not helpful.

Reply:
these PRs were intended to make changes which allows builds by all available (and being used) compilers (i.e. xlC/xlclang/OpenXL), through either cmake or UMA. so, we need to keep these changes anyway.

Reply:
Yes, we would like UMA builds to be possible with OpenXL (although I consider that a bonus, not a necessity), but setting an environment variable for cmake can't benefit UMA builds, so the question remains: why is the variable being set (in cmake)?